packages/server-side-graphql-client/lib/server-side-graphql-client.js

```diff
@@ -0,0 +1,269 @@
+const runQuery = async ({
+  keystone,
+  query,
+  variables,
+  schemaName = 'public',
+  extraContext = {},
+}) => {
+  const { data, errors } = await keystone.executeGraphQL({
+    context: {
+      ...keystone.createContext({ schemaName, authentication: {}, skipAccessControl: true }),
+      ...extraContext,
+    },
+    query,
+    variables,
+  });
+  if (errors) throw errors[0];
+  return data;
+};
+
+const _runChunkedMutation = async ({
+  keystone,
+  query,
+  gqlName,
+  pageSize,
+  items,
+  schemaName,
+  extraContext,
+}) => {
+  if (pageSize <= 0) pageSize = 1;
+
+  const chunks = items.reduce((accum, item, index) => {
+    const chunkIndex = Math.floor(index / pageSize);
+
+    if (!accum[chunkIndex]) {
+      accum[chunkIndex] = [];
+    }
+
+    accum[chunkIndex].push(item);
+
+    return accum;
+  }, []);
+
+  const result = await Promise.all(
+    chunks.map(chunk =>
+      runQuery({ query, variables: { items: chunk }, keystone, schemaName, extraContext })
+    )
+  );
+
+  /*
+   * The result is of the format: [{createUsers: [{id: '123', name: 'aman'}]}, {createUsers: [{id: '456', name: 'mike'}]}].
+   * We need to combine all objects into one array keyed by the `createUsers`, such that, the output is: [{id: '123', name: 'aman'}, {id: '456', name: 'Mike'}]
+   */
+
+  return result.map(item => item[gqlName]).flat(1);
```

Comment:
`.flat()` isn't supported in Node v10, which is our supported version.

Reply:
Oh ok
```suggestion
  return [].concat(result.map(item => item[gqlName]));
```

Reply:
Like this?

Reply:
Yep, except you need to spread the args in.

```suggestion
return [].concat(...result.map(item => item[gqlName]));
```